### ISSUE TITLE
many-to-many query without hard coding

### ISSUE THREAD

**Issue by ninjonxb** [jhund/filterrific#68]
Hi,
I am using filterrific 2.0.5
Ruby 2.1.2
Rails 4.2.0
Given the following models:
    class ItemStatType < ActiveRecord::Base
	    has_many :item_stats
    end
    class ItemStat < ActiveRecord::Base
	    belongs_to :item
	    belongs_to :item_stat_type
    end
    class Item < ActiveRecord::Base
	    has_many :item_stats
    end

ItemStat also has an integer "stat_amount"

Without hard coding I want to do something like this in my view:
    ItemStatType.each do |stat_type|
      <%= stat_type.name %>
        <%= f.text_field(
          :stat_type_grt,
          class: 'filterrific-periodically-observed'
        ) %>
        --
       <%= f.text_field(
          :stat_type_lt,
          class: 'filterrific-periodically-observed'
        ) %>
    end

Basically have a greater than and a less than field for each "ItemStatType".
However I am unable to figure out what the scope would look like for this without hard coding every "ItemStatType" as an individual scope.
The scope would need to work in a way that if there was a value entered in any of the fields it would require that any results match all of the values.
I am still trying to see if I can figure out a solution and I may just end up hard coding it if I have to. But any help would be appreciated.
Thanks Jon
